### PR TITLE
Bug fix for tidl-j6

### PR DESCRIPTION
**Pull request by jianzhong-xu**
Thanks for contributing to TVM!   Please refer to guideline https://tvm.apache.org/docs/contribute/ for useful information and tips. After the pull request is submitted, please request code reviews from [Reviewers](https://github.com/apache/incubator-tvm/blob/master/CONTRIBUTORS.md#reviewers) by @ them in the pull request thread.

### CODE REVIEW

**Comment by jianzhong-xu** [comment]
@trevor-m This PR has a few fixes in TIDL annotation. Please review. Thanks.

File: python/tvm/relay/backend/contrib/tidl.py [hunk 1]
```diff
@@ -1000,15 +980,22 @@ def tidl_import_dense(self, this_node):
         import_lib_dense(num_innodes, num_outnodes, ctypes.c_void_p(weights_array.ctypes.data))
         return True
 
-    def tidl_import_mul(self, this_node):
```
Comment (trevor-m) [inline python/tvm/relay/backend/contrib/tidl.py:1003]: Is tidl_import_mul supposed to be deleted?

File: python/tvm/relay/backend/contrib/tidl.py [hunk 2]
```diff
@@ -1000,15 +980,22 @@ def tidl_import_dense(self, this_node):
...
-        scale = this_node.args[0].data.asnumpy()
-        mul_params.scale = np.amax(scale)
-        import_lib_mul = self.import_lib.tidlImportMul
-        import_lib_mul.argtypes = (ctypes.POINTER(MulParams), ctypes.c_void_p)
-        import_lib_mul.restype = None
-        import_lib_mul(mul_params, ctypes.POINTER(ctypes.c_int)())
+        pooling_params = PoolingParams()
+        pooling_params.kernel_h = pooling_params.kernel_w = 0
+        pooling_params.pad_h = pooling_params.pad_w = 0
+        pooling_params.stride_h = pooling_params.stride_w = 1
```
Comment (trevor-m) [inline python/tvm/relay/backend/contrib/tidl.py:991]: Does kernel width and height of 0 mean something special to the TIDL relay importer?

To directly map max/mean to pool, we would need to do the following (assuming NHWC):
1. If axis == [1]: kernel_h = args[0].checked_type.shape[1], kernel_w = 1
2. If axis == [2]: kernel_h = 1, kernel_w = args[0].checked_type.shape[2]
3. If axis == [1, 2]: kernel_h = args[0].checked_type.shape[1], args[0].checked_type.shape[2]

Then if `keepdims` is false, we need to add a squeeze/reshape.

https://tvm.apache.org/docs/api/python/relay/index.html#tvm.relay.max

File: python/tvm/relay/op/contrib/tidl.py [hunk 3]
```diff
@@ -391,11 +434,13 @@ def _max_pool_whitelist_fn(attrs, args):
...
-def _multiply_whitelist_fn(attrs, args):
-    #supported = True
-    supported = False
-    return supported
+@tvm.ir.register_op_attr("max", "target.tidl")
+def _max_whitelist_fn(attrs, args):
+    return True
+
+@tvm.ir.register_op_attr("mean", "target.tidl")
+def _mean_whitelist_fn(attrs, args):
```
Comment (trevor-m) [inline python/tvm/relay/op/contrib/tidl.py:446]: For max and mean, make sure we check for any attributes the converter doesn't support:
https://tvm.apache.org/docs/api/python/relay/index.html#tvm.relay.max
For example, we probably don't support when exclude is True and you may not support keepdims. It also isn't possible to support the case where axis is the channels_dim when mapping to pooling.

**Review by trevor-m — CHANGES_REQUESTED** [review]
Thanks Jianzhong!

File: python/tvm/relay/backend/contrib/tidl.py (hunk 2 above)
Comment (jianzhong-xu) [reply python/tvm/relay/backend/contrib/tidl.py:991]: Yes, setting kernel width and height to 0 means this is a global average/max pool for TIDL. 
TIDL doesn't support keepdims being false. I will correct the annotation part.

File: python/tvm/relay/op/contrib/tidl.py (hunk 3 above)
Comment (jianzhong-xu) [reply python/tvm/relay/op/contrib/tidl.py:446]: I will add these checks. Thanks!

File: python/tvm/relay/backend/contrib/tidl.py (hunk 1 above)
Comment (jianzhong-xu) [reply python/tvm/relay/backend/contrib/tidl.py:1003]: Yes, "multiply" is not supported by this version of TIDL.

File: python/tvm/relay/op/contrib/tidl.py (hunk 3 above)
Comment (jianzhong-xu) [reply python/tvm/relay/op/contrib/tidl.py:446]: TIDL supports "max" and "mean" for both keepdims being True and False.

**Review by trevor-m — APPROVED** [review]

**Pull request merged**